kernel/arch/dreamcast/include/dc/biosfont.h

```diff
@@ -26,267 +31,296 @@
-#define BFONT_UPARROW               BFONT_DREAMCAST_SPECIFIC+(3*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_DOWNARROW             BFONT_DREAMCAST_SPECIFIC+(4*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_LEFTARROW             BFONT_DREAMCAST_SPECIFIC+(5*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_RIGHTARROW            BFONT_DREAMCAST_SPECIFIC+(6*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_UPRIGHTARROW          BFONT_DREAMCAST_SPECIFIC+(7*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_DOWNRIGHTARROW        BFONT_DREAMCAST_SPECIFIC+(8*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_DOWNLEFTARROW         BFONT_DREAMCAST_SPECIFIC+(9*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_UPLEFTARROW           BFONT_DREAMCAST_SPECIFIC+(10*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_ABUTTON               BFONT_DREAMCAST_SPECIFIC+(11*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_BBUTTON               BFONT_DREAMCAST_SPECIFIC+(12*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_CBUTTON               BFONT_DREAMCAST_SPECIFIC+(13*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_DBUTTON               BFONT_DREAMCAST_SPECIFIC+(14*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_XBUTTON               BFONT_DREAMCAST_SPECIFIC+(15*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_YBUTTON               BFONT_DREAMCAST_SPECIFIC+(16*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_ZBUTTON               BFONT_DREAMCAST_SPECIFIC+(17*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_LTRIGGER              BFONT_DREAMCAST_SPECIFIC+(18*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_RTRIGGER              BFONT_DREAMCAST_SPECIFIC+(19*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_STARTBUTTON           BFONT_DREAMCAST_SPECIFIC+(20*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
-#define BFONT_VMUICON               BFONT_DREAMCAST_SPECIFIC+(21*BFONT_WIDE_WIDTH*BFONT_HEIGHT/8)
+/** \brief Size of a row for JISX-0208 characters */
+#define JISX_0208_ROW_SIZE          94
+/** \brief Start of Wide Characters in Font Block */
+#define BFONT_WIDE_START            (288 * BFONT_THIN_WIDTH * BFONT_HEIGHT / 8)
+/** \brief Start of JISX-0208 Rows 1-7 in Font Block */   
+#define BFONT_JISX_0208_ROW1        BFONT_WIDE_START
+/** \brief Start of JISX-0208 Row 16-47 (Start of Level 1) in Font Block */   
+#define BFONT_JISX_0208_ROW16       BFONT_WIDE_START + (658 * BFONT_WIDE_WIDTH * BFONT_HEIGHT / 8)  
+/** \brief JISX-0208 Row 48-84 (Start of Level 2) in Font Block */
+#define BFONT_JISX_0208_ROW48       BFONT_JISX_0208_ROW16 + ((32 * JISX_0208_ROW_SIZE) * BFONT_WIDE_WIDTH * BFONT_HEIGHT / 8) 
+
+/** \brief Start of DC Specific Characters in Font Block */
+#define BFONT_DREAMCAST_SPECIFIC    BFONT_WIDE_START + (7056 * BFONT_WIDE_WIDTH * BFONT_HEIGHT / 8) 
+/** \brief Takes a DC-specific icon index and returns a character offset. */
+#define BFONT_DC_ICON(offset)       BFONT_DREAMCAST_SPECIFIC + ( (offset) * BFONT_WIDE_WIDTH * BFONT_HEIGHT / 8)
```

Comment:
While you're macro-ing ...`BFONT_WIDE_WIDTH * BFONT_HEIGHT / 8` would  be a good candidate. But if there's a reason to leave it out, that's fine too. bytes-per-char.